### PR TITLE
Added books on computer architecture in Hindi. Created a topic in index for C which was missing. 

### PR DESCRIPTION
**Pull request by CharithaCS**
## What does this PR do?
Added books and resources on computer architecture in Hindi. Also created a topic in index for C which was missing earlier.

### Description
This book covers essential topics in computer system architecture, including digital number systems, Boolean algebra, logic gates, sequential and combinational circuits, registers, counters, computer memory, CPU architecture, and more. Understanding these concepts is fundamental for computer science and engineering students. It lays the foundation for designing and programming computers, from basic logic gates to complex microprocessors. This knowledge enables efficient circuit design, data manipulation, and program execution. Moreover, it aids in troubleshooting and optimizing computer systems. In a technology-driven world, this book is indispensable for those pursuing careers in hardware design, software development, and computer engineering, providing a strong grasp of how computers function at a fundamental level.

### Why is this valuable (or not)?
These supplements are of great significance as they offer valuable learning resources for those interested in delving into Computer architecture and organization, specifically to people who are more acquainted with Hindi, ensuring access to a wide range of people. Many modern devices, from smartphones to IoT devices, rely on embedded systems. Understanding computer architecture is essential for designing and programming these systems.

### How do we know it's really free?
The provided links grant free access to the books, and there are no signs of any accompanying charges or limitations.

### For book lists, is it a book? For course lists, is it a course? etc.
They are books

## Checklist:
- [*] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [*] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [*] Include author(s) and platform where appropriate.
- [*] Put lists in alphabetical order, correct spacing.
- [*] Add needed indications (PDF, access notes, under construction).
- [*] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-hi.md [hunk 1]
```diff
@@ -23,6 +25,13 @@
 * [C++ Introduction Book \| Hindi](https://ncsmindia.com/wp-content/uploads/2012/04/c++-hindi.pdf) - NCMS India (PDF)
 
 
+### Computer architecture
+
+* [कम्प्यूटर सिस्टम आर्किटेक्चर](https://www.aicte-india.org/sites/default/files/HINDI_BOOKS/BOOK%207.pdf) - डॉ. एस. एस. श्रीवास्तव, उच्च शिक्षा उत्कृष्टता संस्थान, भोपाल (म. प्र. ) (PDF)
+* [Computer architecture and organization notes](https://drive.google.com/file/d/1aE0mIBpmjansb9onxpw4Fh9xfGYPI96B/view) - eHindiStudy.com
```
Comment (eshellman) [inline books/free-programming-books-hi.md:31]: In general, we do not link to file sharing sites, or file hosting sites such as google docs. or dropbox. remove it

File: books/free-programming-books-hi.md [hunk 2]
```diff
@@ -23,6 +25,13 @@
...
+* [Objective questions on Computer Organization and architecture](https://testbook.com/objective-questions/hn/mcq-on-computer-organization-and-architecture--5eea6a0a39140f30f369dacc) - TestBook.com
```
Comment (eshellman) [inline books/free-programming-books-hi.md:32]: This is not really a book, it's an exam site and all the downloads require a cell phone number so not free. Also a really annoying ad every time I click

**Review by eshellman — CHANGES_REQUESTED** [review]
First book is OK. Remove the others. We don't include honorifics (Dr., Prof. etc.) I'm not sure if your attribution has honorifics in Hindu.

**Comment by CharithaCS** [comment]
Thank you very much for the clarifications. I've retained only the first one and removed the honorifics. Please check.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**